theano/tensor/nnet/opt.py

```diff
@@ -175,6 +181,7 @@ def local_conv2d_gradweight_cpu(node):
         rval = rval[:, :, ::-1, ::-1]
         rval = theano.tensor.patternbroadcast(rval,
                                               node.outputs[0].broadcastable)
+        copy_stack_trace(node.outputs[0], rval)
```

Comment:
I would keep this one, but also add the stack trace to the output of `convGrad3D`, defined on l.172.